src/TicketingSystem.BusinessLogic/Services/PaymentService.cs

```diff
@@ -15,51 +16,69 @@ namespace TicketingSystem.BusinessLogic.Services
     public class PaymentService(IMongoRepository<Payment> repository, IMapper mapper)
         : GenericEntityService<Payment, PaymentDto>(repository, mapper), IPaymentService
     {
-        public Task UpdatePaymentState(string paymentId, PaymentState state)
+        public async Task UpdatePaymentState(string paymentId, PaymentState state, CancellationToken cancellationToken = default)
         {
-            return _repository.UpdateAsync(paymentId, p => p.State, _mapper.Map<PaymentState>(state));
+            var payment = await _repository.GetByIdAsync(paymentId, cancellationToken);
+
+            if (state is PaymentState.Completed or PaymentState.Failed)
+            {
```

Comment:
I have reverted this condition to if (state is not (PaymentState.Completed or PaymentState.Failed))... and tests are still green

Reply:
Changed the condition to clean CartItems only in case of Completion or Fail of the payment and leave there a single update execution - that way the tests started to fail